skyportal/models.py

```diff
@@ -80,14 +82,26 @@ class Stream(Base):
 User.groups = relationship('Group', secondary='group_users',
                            back_populates='users')
 
+
 @property
 def token_groups(self):
     return self.created_by.groups
+
+
 Token.groups = token_groups
 
 
 class Source(Base):
```

Comment:
I think we should implement delete protection and other logic as a method of `Source`. For example, if `is_source` is True then the row cannot be deleted by a vaccuming process that is trying to reap stale candidates. Also, at the time that `is_source` is set to True, then `is_candidate` should be set to False.

Reply:
That'll be baked into the deletion query - `filter(Source.is_source.is_(False)).filter(Source.is_candidate.is_(True))`

Reply:
As for the latter part, my understanding is that oftentimes we'd want both `is_candidate` and `is_source` to be `True`. If a source has been previously saved for some other group you're not a member of, you still want to see the associated candidate on your scanning page, and then want to be able to click on the ID and see the candidate view page